app/gui2/src/components/lexical/MarkdownEditorImpl.vue

```diff
@@ -0,0 +1,91 @@
+<script setup lang="ts">
+import EditorRoot from '@/components/lexical/EditorRoot.vue'
+import MarkdownSync from '@/components/lexical/MarkdownSync.vue'
+import { useFocusDelayed } from '@/composables/focus'
+import { CodeHighlightNode, CodeNode } from '@lexical/code'
+import { AutoLinkNode, LinkNode } from '@lexical/link'
+import { ListItemNode, ListNode } from '@lexical/list'
+import { TRANSFORMERS } from '@lexical/markdown'
+import { HeadingNode, QuoteNode } from '@lexical/rich-text'
+import { TableCellNode, TableNode, TableRowNode } from '@lexical/table'
+import { syncRef } from '@vueuse/core'
+import {
+  LexicalComposer,
+  LexicalContentEditable,
+  LexicalMarkdownShortcutPlugin,
+  LexicalRichTextPlugin,
+} from 'lexical-vue'
+import { ref } from 'vue'
+
+const text = defineModel<string>({ required: true })
+const focused = defineModel<boolean>('focused', { default: false })
+
+const root = ref<HTMLElement>()
+
+const config = {
+  editable: true,
+  namespace: 'MarkdownEditor',
+  theme: {},
+  nodes: [
+    HeadingNode,
+    QuoteNode,
+    ListItemNode,
+    ListNode,
+    AutoLinkNode,
+    LinkNode,
+    CodeHighlightNode,
+    CodeNode,
+    TableCellNode,
+    TableNode,
+    TableRowNode,
+  ],
+}
+
+syncRef(focused, useFocusDelayed(root).focused)
+</script>
+
+<template>
+  <LexicalComposer :initial-config="config" @error="console.error($event)">
+    <LexicalRichTextPlugin>
+      <template #contentEditable>
+        <LexicalContentEditable
+          :spellcheck="false"
+          class="lexicalContent"
+          @wheel.stop
+          @contextmenu.stop
+        />
+      </template>
+    </LexicalRichTextPlugin>
+    <MarkdownSync v-model="text" />
+    <EditorRoot v-model="root" />
+    <LexicalMarkdownShortcutPlugin :transformers="TRANSFORMERS" />
+  </LexicalComposer>
+</template>
+
+<style scoped>
+.lexicalContent {
+  height: 100%;
+  outline: none;
+}
+</style>
+
+<style>
+h1 {
+  font-weight: 700;
+  font-size: 16px;
+  line-height: 1.75;
+}
+
+h2,
```

Comment:
Are those styles actually functioning? Because those are defined in scoped style block, they will only apply to vue-owned DOM nodes by default.

Reply:
They are in an unscoped block.

Reply:
Right, I missed that. In this case I think the styles are actually too general, likely also influencing the dahsboard as well by accident. Ideally this would still be in scoped block, but within a [`:deep` subselector](https://vuejs.org/api/sfc-css-features.html#deep-selectors), so it is scoped to the contents of this component.